core/services/blockhashstore/delegate.go

```diff
@@ -0,0 +1,178 @@
+		return nil, errors.Errorf(
+			"blockhashstore.Delegate expects a BlockhashStoreSpec to be present, got %+v", jb)
+	}
+
+	chain, err := d.chains.Get(jb.BlockhashStoreSpec.EVMChainID.ToInt())
+	if err != nil {
+		return nil, fmt.Errorf(
+			"getting chain ID %d: %w", jb.BlockhashStoreSpec.EVMChainID.ToInt(), err)
+	}
+
+	keys, err := d.ks.SendingKeys()
+	if err != nil {
+		return nil, errors.Wrap(err, "getting sending keys")
+	}
+	fromAddress := keys[0].Address
+	if jb.BlockhashStoreSpec.FromAddress != nil {
+		fromAddress = *jb.BlockhashStoreSpec.FromAddress
+	}
+
+	bhs, err := blockhash_store.NewBlockhashStore(
+		jb.BlockhashStoreSpec.BlockhashStoreAddress.Address(), chain.Client())
+	if err != nil {
+		return nil, errors.Wrap(err, "building BHS")
+	}
+
+	var coordinators []Coordinator
+	if jb.BlockhashStoreSpec.CoordinatorV1Address != nil {
+		var c *v1.VRFCoordinator
+		if c, err = v1.NewVRFCoordinator(
+			jb.BlockhashStoreSpec.CoordinatorV1Address.Address(), chain.Client()); err != nil {
+
+			return nil, errors.Wrap(err, "building V1 coordinator")
+		}
+		coordinators = append(coordinators, NewV1Coordinator(c))
+	}
+	if jb.BlockhashStoreSpec.CoordinatorV2Address != nil {
+		var c *v2.VRFCoordinatorV2
+		if c, err = v2.NewVRFCoordinatorV2(
+			jb.BlockhashStoreSpec.CoordinatorV2Address.Address(), chain.Client()); err != nil {
+
+			return nil, errors.Wrap(err, "building V2 coordinator")
+		}
+		coordinators = append(coordinators, NewV2Coordinator(c))
+	}
+
+	bpBHS, err := NewBulletproofBHS(chain.Config(), fromAddress.Address(), chain.TxManager(), bhs)
+	if err != nil {
+		return nil, errors.Wrap(err, "building bulletproof bhs")
+	}
+
+	log := d.logger.Named("BHS Feeder").With("jobID", jb.ID, "externalJobID", jb.ExternalJobID)
+	feeder := NewFeeder(
+		log,
+		NewMultiCoordinator(coordinators...),
+		bpBHS,
+		int(jb.BlockhashStoreSpec.WaitBlocks),
+		int(jb.BlockhashStoreSpec.LookbackBlocks),
+		func(ctx context.Context) (uint64, error) {
+			head, err := chain.Client().HeadByNumber(ctx, nil)
+			if err != nil {
+				return 0, errors.Wrap(err, "getting chain head")
+			}
+			return uint64(head.Number), nil
+		})
+
+	return []job.Service{&service{
+		feeder:     feeder,
+		pollPeriod: jb.BlockhashStoreSpec.PollPeriod,
+		logger:     log,
+	}}, nil
+}
+
+// AfterJobCreated satisfies the job.Delegate interface.
+func (d *Delegate) AfterJobCreated(spec job.Job) {}
+
+// BeforeJobDeleted satisfies the job.Delegate interface.
+func (d *Delegate) BeforeJobDeleted(spec job.Job) {}
+
+// service is a job.Service that runs the BHS feeder every pollPeriod.
+type service struct {
+	feeder     *Feeder
+	stop       chan struct{}
+	pollPeriod time.Duration
+	logger     logger.Logger
+	parentCtx  context.Context
+	cancel     context.CancelFunc
+}
+
+// Start the BHS feeder service, satisfying the job.Service interface.
+func (s *service) Start() error {
+	if s.stop != nil {
+		return errors.New("service already started")
+	}
```

Comment:
Minor: Might be simpler to embed `utils.StartStopOnce` in `service` so that you get the once functionality both ways.